index.js

```diff
@@ -1,31 +1,128 @@
 // 1.Fibonacci
 
-// function fibo(n) {
-//   return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
-// }
-// fibo(5);
-// fibo(6);
-// fibo(7);
+function fibo(n) {
+  return n <= 1 ? n : fibo(n - 1) + fibo(n - 2);
+}
+fibo(5);
+fibo(6);
+fibo(7);
 
 // 2. Implementing with reduce
 
 //concat;
+const arr = [1, 2];
+const arr1 = [3, 4];
+const arr2 = 5;
+
+const concatFn = (array, ...params) =>
+  params.reduce(
+    (acc, newEl) =>
+      Array.isArray(newEl) ? [...acc, ...newEl] : [...acc, newEl],
+    array
+  );
+
+console.log(concatFn(arr, arr1));
+console.log(concatFn(arr1, arr2));
+console.log(`My new ${arr}`);
+
+//filter and map;
+
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+
+const filter = (newArr, fn) =>
+  newArr.reduce((acc, person) => (fn(person) ? [...acc, person] : acc), []);
+
+let fn1 = (obj) => {
+  return obj.age > 18;
+};
+
+console.log(filter(friends, fn1));
 
-//filter;
 // find;
+let friends = [
+  {
+    name: "John",
+    age: 15,
+  },
+  {
+    name: "Michael",
+    age: 25,
+  },
+  {
+    name: "Lisa",
+    age: 30,
+  },
+];
+const findLisa = (newArr, fn) =>
```

Comment:
now Add in array one more person, that is 30 years old and try.

find must return first match